packages/ember-toucan-form/src/-private/textarea-field.ts

```diff
@@ -0,0 +1,33 @@
+import Component from '@glimmer/component';
+
+import type { ToucanFormTextareaFieldComponentSignature as BaseTextareaFieldSignature } from '@crowdstrike/ember-toucan-core/components/form/fields/textarea';
+
+type ComponentArguments = BaseTextareaFieldSignature['Args'] & {
+  name: string;
+
+  // TODO: We need to type this properly
+  // What is the type when we are a component?
+  // Is this possible with TS+Glint today?
+  form: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    Field: any;
+  };
```

Comment:
@simonihmig do you know how we can properly type this component from `ember-headless-form`?  Maybe a few issues:

1. I'm not sure how to declare this argument as a component as it's received from the parent component.  I've tried a few different syntax options like `Field: [{ value?: string }]`, etc., but Glint complains about it wanting to be an `Invokable` as it's a component.
2. The `Field` [type is exported, but under `/-private` ](https://github.com/CrowdStrike/ember-headless-form/blob/main/packages/ember-headless-form/src/-private/components/field.ts#L27).  Should I make a PR to export this as a public interface/type so we can import it and use it here? Or should we copy/paste the types over?

I very well could be missing something silly here!

Reply:
I think this has to match what the `<HeadlessForm>` component yields [here](https://github.com/CrowdStrike/ember-headless-form/blob/main/packages/ember-headless-form/src/components/headless-form.ts#L89-L128) - and we shouldn't try to redefine it, instead reuse the types that headless-form already provides. However I am not sure how to only get the yielded block, when the only public type that we can import is that of the [`HeadlessForm` component](https://github.com/CrowdStrike/ember-headless-form/blob/main/packages/ember-headless-form/src/index.ts#L10). A Glint helper like `GetBlockParams(HeadlessForm, 'default')[0]` would be nice, but it doesn't seem to exist...

Maybe we can build such a helper, but I cannot do this off the top of my head, without actually working on the code 😅. I can give it a shot tomorrow maybe if you want? Not sure if my TS skills will be sufficient though...

Reply:
Oh that'd be pretty cool if we could make that work!  If you have some time tomorrow, that'd be great.  I'll also be on in the AM if you'd like to pair and see if we can arrive at something.  Maybe with our powers combined we can figure this out! 🎉